### PR TITLE
Topics are now db.Owned and filter is by ID

### PR DESCRIPTION
**Pull request by quaxsze**
fix https://github.com/etalab/data.gouv.fr/issues/1177
fix https://github.com/etalab/data.gouv.fr/issues/1178

### CODE REVIEW

**Comment by quaxsze** [comment]
No migration is needed as topics already had an `owner` referenceField.
Reindex is needed to index the ids instead of the slugs

File: udata/core/dataset/search.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def serialize(cls, dataset):
             'owner': str(owner.id) if owner else None,
             'format': [r.format.lower() for r in dataset.resources if r.format],
             'schema': [r.schema.get('name') for r in dataset.resources if r.schema],
-            'topics': [t.slug for t in topics if topics],
+            'topics': [t.id for t in topics if topics],
```
Comment (maudetes) [inline udata/core/dataset/search.py:101]:
```suggestion
            'topics': [str(t.id) for t in topics if topics],
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 - Return 400 instead of 500 in case of not ObjectID arg in API [#2889](https://github.com/opendatateam/udata/pull/2889)
 - Use a single session for reindex [#2891](https://github.com/opendatateam/udata/pull/2891)
 - Topics creation, update and deletion are now opened to all users [#2898](https://github.com/opendatateam/udata/pull/2898)
+- Topics are now `db.Owned` and searchable by `id` in dataset search [#2901](https://github.com/opendatateam/udata/pull/2901)
```
Comment (maudetes) [inline CHANGELOG.md:9]: You could probably merge with the previous Topics entry?

**Review by maudetes — COMMENTED** [review]
Seems to work quite easily, well done :)

Could you maybe add a test on topic creation as an organization (probably similar to tests for reuse)?

File: udata/tests/api/test_topics_api.py [hunk 3]
```diff
@@ -42,6 +44,24 @@ def test_topic_api_create(self):
...
+    def test_topic_api_create_as_org(self):
+        '''It should create a topic as organization from the API'''
+        data = TopicFactory.as_dict()
+        data['datasets'] = [str(d.id) for d in data['datasets']]
+        data['reuses'] = [str(r.id) for r in data['reuses']]
+        user = self.login()
+        member = Member(user=user, role='editor')
+        org = OrganizationFactory(members=[member])
+        data['organization'] = str(org.id)
+        print(data)
```
Comment (maudetes) [inline udata/tests/api/test_topics_api.py:56]:
```suggestion
```

**Review by maudetes — APPROVED** [review]
:pray:

**Pull request merged**